src/transport/raw/tests/TestTCP.cpp

```diff
@@ -67,6 +66,11 @@ constexpr uint32_t kMessageCounter  = 18;
 const char PAYLOAD[]          = "Hello!";
 const char messageSize_TEST[] = "\x00\x00\x00\x00";
 
+uint16_t GetRandomPort()
+{
+    return static_cast<uint16_t>(CHIP_PORT + chip::Crypto::GetRandU16() % 100);
```

Comment:
Can we use 0 to get a port auto-assigned? We still seem to play a lottery (but maybe we smaller chances of failures).

Reply:
We need to add some code to fetch the port assigned by the kernel for TCP so that it can be used in other parts of the tests, e.g., the Client side, etc. We can look at this in a future PR for the tests. The chances of _consecutive_ random numbers being equal should be extremely small. Test cases that are slightly farther apart are fine even if they collide.
But that aside, we cannot do the same for the regular server code because of the way the TransportMgr initializes all the transports together in the tuple. Because the DNS-SD advertisements pick only index 0, the TCP port should match the UDP port(whatever it is). Passing 0 for each transport Bind() will make them different.

Reply:
Please create a TODO issue and reference it here in a comment.

Reply:
If we need consecutive tests to be different, we could have a `cache` for our random ports and ensure maybe that 3 consecutive calls to this method never return the same value (i.e. keep randomizing until we get a previously unused one)? Or maybe just make the calls to this method rotate over a known set of values without any random?

That seems maybe even better ...

Reply:
How about `GetPort` keeps returning a new value with a index % 100 ... so it restarts every 100 calls, but at least 100 consecutive tests will get different ports. Would that cause issues?

Reply:
Filed Todo: https://github.com/project-chip/connectedhomeip/issues/37155